src/lib.rs

```diff
@@ -106,6 +109,57 @@ pub enum ReturnIndent {
 
 impl_enum_decodable!(ReturnIndent, WithArgs, WithWhereClause);
 
+enum ErrorKind {
+    // Line has more than config!(max_width) characters
+    LineOverflow,
+    // Line ends in whitespace
+    TrailingWhitespace,
+    // TO-DO item in comments without an issue number
+    UnnumberedTodo,
+    // FIX-ME item in comments without an issue number
+    UnnumberedFixme,
+}
+
+impl<'a> fmt::Display for ErrorKind {
+    // Prints all the formatting errors.
+    fn fmt(&self, fmt: &mut fmt::Formatter) -> Result<(), fmt::Error> {
+        let msg = match *self {
+            ErrorKind::LineOverflow => "line exceeded maximum length",
+            ErrorKind::TrailingWhitespace => "left trailing whitespace",
+            ErrorKind::UnnumberedTodo => "found TODO without issue number",
+            ErrorKind::UnnumberedFixme => "found FIXME without issue number"
+        };
+
+        write!(fmt, "{}", msg)
+    }
+}
+
+// Formatting errors that are identified *after* rustfmt has run
+struct FormattingError {
+    line: u32,
+    kind: ErrorKind,
+}
+
+struct FormatReport {
+    file_error_map: HashMap<String, Vec<FormattingError>>,
```

Comment:
Could you comment on this field to explain what the String key in the hash map is